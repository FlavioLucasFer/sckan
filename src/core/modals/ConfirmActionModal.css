.confirm-action-modal-height {
	height: 200px !important;
	width: 700px !important;
}