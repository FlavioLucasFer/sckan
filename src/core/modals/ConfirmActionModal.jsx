import './ConfirmActionModal.css';
import React from 'react';
import PropTypes from 'prop-types';

import Modal from 'core/components/Modal';

ConfirmActionModal.propTypes = {
	title: PropTypes.string.isRequired,
	id: PropTypes.string.isRequired,
	onConfirm: PropTypes.func.isRequired,
	onCancel: PropTypes.func,
};

function ConfirmActionModal(props) {
	const {
		title,
		id,
		onConfirm,
		onCancel,
	} = props;

	const actions = [{
		id: 'confirm-action',
		title: 'Yes',
		className: 'modal-close waves-green',
		onClick: () => onConfirm(),
	}, {
		id: 'deny-action',
		title: 'No',
		className: 'modal-close waves-red',
		onClick: () => onCancel && onCancel(),
	}];

	return (
		<Modal id={id} 
			title={title} 
			actions={actions}
			className="confirm-action-modal-height" />
	);
}

export default ConfirmActionModal;