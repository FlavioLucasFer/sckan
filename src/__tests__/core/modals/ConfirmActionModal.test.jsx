import React from 'react';
import { shallow } from 'enzyme';

import ConfirmActionModal from 'core/modals/ConfirmActionModal';

describe('Testing ConfirmActionModal core modal component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<ConfirmActionModal title="confirm action modal test"
				id="confirm-action-modal-test"
				onConfirm={() => console.log('on confirm test')} />
		);

		expect(wrapper.debug()).toMatchSnapshot();
		
		wrapper.setProps({
			title: 'testing',
			onCancel: () => console.log('on cancel test'),
		});
	
		expect(wrapper.debug()).toMatchSnapshot();
	});
});