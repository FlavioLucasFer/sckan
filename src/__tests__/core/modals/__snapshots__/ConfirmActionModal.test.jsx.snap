// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmActionModal core modal component should render correctly 1`] = `"<Modal id=\\"confirm-action-modal-test\\" title=\\"confirm action modal test\\" actions={{...}} className=\\"confirm-action-modal-height\\" />"`;

exports[`Testing ConfirmActionModal core modal component should render correctly 2`] = `"<Modal id=\\"confirm-action-modal-test\\" title=\\"testing\\" actions={{...}} className=\\"confirm-action-modal-height\\" />"`;
